File: test_parse.py
import re

with open('index.html', 'r', encoding='utf-8') as f:
    content = f.read()

scripts = re.findall(r'<script>(.*?)</script>', content, re.DOTALL)
for i, script in enumerate(scripts):
    if 'AIEngine' in script:
        with open(f'extracted_engine_{i}.js', 'w', encoding='utf-8') as f:
            f.write(script)
        print(f"Extracted script {i} containing AIEngine.")

File: test_parse.js
const fs = require('fs');
const content = fs.readFileSync('index.html', 'utf8');

const regex = /<script>([\s\S]*?)<\/script>/g;
let match;
while ((match = regex.exec(content)) !== null) {
    const scriptBody = match[1];
    if (scriptBody.includes('AIEngine')) {
        fs.writeFileSync('extracted_engine.js', scriptBody);
        console.log("Extracted script containing AIEngine.");
    }
}
